lib/rules/selector-not-notation/index.mjs

```diff
@@ -82,43 +85,38 @@ const rule = (primary, _, context) => {
 			selectorRoot.walkPseudos((pseudo) => {
 				if (!isNot(pseudo)) return;
 
+				const range = getRange(pseudo, ruleNode);
+				let fix;
+
 				if (primary === 'complex') {
 					const prev = pseudo.prev();
 					const hasConsecutiveNot = prev && isNot(prev);
 
 					if (!hasConsecutiveNot) return;
 
-					if (context.fix) return fixComplex(prev);
+					fix = fixComplex.bind({ ruleNode, selectorRoot }, prev);
```

Comment:
[suggest] Using `bind` and `this` is a bit too tricky. Instead, how about just creating a function like this?

```js
fix = () => {
	fixComplex(prev);
	ruleNode.selector = selectorRoot.toString();
	return getNewRange(prev, ruleNode);
};
```